### PR TITLE
Use `font_size` setting to control `napari-console` font-size

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

See napari/napari-console#32 and napari/napari-console#33

# Description

Enable `napari-console` to react to the changes over the font size setting from napari. This needs some changes over `napari-console` too. See napari/napari-console#33

A preview:

console_font_size

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
@dalthviz what's the proper/preferred order of merging this and the console PR?

**Comment by dalthviz** [comment]
Hi there, I think the `napari-console` PR should be merged first. In fact I think a release with the change will be needed so this PR can update the `napari-console` dependency constraint (something like `napari-console >=0.0.10`).

However, thinking about this and checking the current constraint for `napari-console`, I think the work done over the console side could break things if used with napari 0.4.x . I think part of the code I removed there needs to still be there to keep `napari-console` compatibility with napari 0.4.x 🤔 (enable the console to update its style sheet). For the moment, I will mark both PRs as WIP, do some tests, and come back to you after I have ensured changes keep compatibility between the console and previous napari versions :+1:

**Comment by dalthviz** [comment]
I added some changes to the console PR and checked locally with napari 0.4.19 so things seem ready for review now :+1:

File: napari/_qt/qt_main_window.py
Comment (Czaki) [inline napari/_qt/qt_main_window.py:1563]: `_console` could be `None` and it is intentional

**Comment by DragaDoncila** [comment]
@dalthviz tests are failing here because of the napari-console requirement. Are we waiting on console to release before this PR can be updated?

**Comment by dalthviz** [comment]
> Are we waiting on console to release before this PR can be updated?

Correct, in fact https://github.com/napari/napari-console/pull/33 needs to be merged first and then a release with that change be done. However, I'm not totally sure if this is the right approach to follow/if there are alternatives to achieve the behavior this work is doing. Maybe these PRs (this one and https://github.com/napari/napari-console/pull/33) just should be closed

**Comment by psobolewskiPhD** [comment]
I think it's this -- linking napari font size with the console -- or making people more aware that the console has qtconsole keybinds, so you can use the zoom in/out keybinds to change the font in the console:
https://github.com/napari/napari-console/issues/32#issuecomment-1997831720

**Comment by jni** [comment]
@psobolewskiPhD I would expect you to have a preference for either option?

I think using the preference is good because it'll remember it between sessions, whereas I expect using the shortcut doesn't?

**Review by Czaki — APPROVED** [review]

File: pyproject.toml
Comment (Czaki) [inline pyproject.toml]:
```suggestion
    "napari-console>=0.1.1",
```

**Comment by Czaki** [comment]
Once #7287 is merged, we could iterate this PR again.

**Review by Czaki — APPROVED** [review]

**Comment by jni** [comment]
Amazing! 🚀

**Pull request merged**
